File: benchmarks/string_ops.py
#!/usr/bin/env python3
import time

def string_operations():
    text = "Hello World from SKY Programming Language"
    iterations = 100000
    
    start = time.time()
    
    for i in range(iterations):
        upper = text.upper()
        lower = text.lower()
        replaced = text.replace("SKY", "GO")
        joined = "-".join(["a", "b", "c"])
        split_result = text.split(" ")
    
    end = time.time()
    duration = (end - start) * 1000  # Convert to milliseconds
    
    print(f"Python String operations ({iterations} iterations)")
    print(f"Duration: {duration:.2f} ms")

def main():
    string_operations()

if __name__ == "__main__":
    main()

File: benchmarks/string_ops.js
#!/usr/bin/env node

function stringOperations() {
    const text = "Hello World from SKY Programming Language";
    const iterations = 100000;
    
    const start = Date.now();
    
    for (let i = 0; i < iterations; i++) {
        const upper = text.toUpperCase();
        const lower = text.toLowerCase();
        const replaced = text.replace("SKY", "GO");
        const joined = ["a", "b", "c"].join("-");
        const splitResult = text.split(" ");
    }
    
    const end = Date.now();
    const duration = end - start;
    
    console.log(`JavaScript String operations (${iterations} iterations)`);
    console.log(`Duration: ${duration} ms`);
}

function main() {
    stringOperations();
}

main();
